File: praca-inzynierska-frontend/src/app/app/model/dto/EmployeeDto.ts
import { AccountDto } from './AccountDto'
import { EmployeeStatus } from '../EmployeeStatus' 
import { PositionDto } from './PositionDto'

export class EmployeeDto {
    constructor(
        public employeeId: number,
        public account: AccountDto,
        public employeeStatus: EmployeeStatus,
        public positions: PositionDto[],
        public invitationToken: string
    ) {}

    static compare = (a: EmployeeDto, b: EmployeeDto) => {
        let c = a.account.lastName.localeCompare(b.account.lastName)
        return c != 0 ? c : a.account.firstName.localeCompare(b.account.firstName)
    }
}

File: praca-inzynierska-backend/application/src/main/kotlin/jwer/pracainzynierskabackend/model/dto/EmployeeDto.kt
package jwer.pracainzynierskabackend.model.dto

import jwer.pracainzynierskabackend.model.auth.AccountType
import jwer.pracainzynierskabackend.model.entity.Employee
import jwer.pracainzynierskabackend.model.entity.EmployeeStatus
import java.time.LocalDate

data class EmployeeDto(
        val employeeId: Long,
        val account: AccountDto,
        val employeeStatus: EmployeeStatus,
        val positions: List<PositionDto>,
        val invitationToken: String?
) {
    constructor(e: Employee) : this(
            e.id,
            AccountDto(e.account, AccountType.EMPLOYEE),
            e.employeeStatus,
            e.positions.map { PositionDto(it.id, it.name) },
            e.invitationToken
    )
}